pkg/conjurpolicy/types.go

```diff
@@ -248,33 +156,42 @@ func (s *PolicyStatements) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
-type Policy struct {
-	Resource    `yaml:"-"`
-	Id          string           `yaml:"id"`
-	Annotations Annotations      `yaml:"annotations,omitempty"`
-	Owner       ResourceRef      `yaml:"owner,omitempty"`
-	Body        PolicyStatements `yaml:"body,omitempty"`
+func (p Policy) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(p, KindPolicy)
 }
 
-type Layer struct {
-	Resource `yaml:"-"`
+func (v Variable) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(v, KindVariable)
 }
 
-type Grant struct {
-	Resource `yaml:"-"`
-	Role     ResourceRef `yaml:"role"`
-	Member   ResourceRef `yaml:"member"`
+func (u User) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(u, KindUser)
 }
 
-type Host struct {
-	Resource    `yaml:"-"`
-	Id          string            `yaml:"id"`
-	Owner       ResourceRef       `yaml:"owner,omitempty"`
-	Body        PolicyStatements  `yaml:"body,omitempty"`
-	Annotations map[string]string `yaml:"annotations,omitempty"`
+func (g Group) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(g, KindGroup)
 }
 
-type Delete struct {
-	Resource `yaml:"-"`
-	Record   ResourceRef `yaml:"record"`
+func (l Layer) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(l, KindLayer)
+}
+
+func (g Grant) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(g, KindGrant)
+}
+
+func (h Host) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(h, KindHost)
+}
+
+func (d Delete) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(d, KindDelete)
+}
+
+func (p Permit) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(p, KindPermit)
+}
+
+func (d Deny) MarshalYAML() (interface{}, error) {
+	return MarshalYAMLWithTag(d, KindDeny)
```

Comment:
TODO: Probably something to be considered separate from this PR... it would be nice if all of this was just code generated much like the enums are.